src/goto-programs/goto_trace.cpp

```diff
@@ -61,61 +62,45 @@ void goto_trace_stept::output(
     UNREACHABLE;
   }
 
-  if(type==typet::ASSERT || type==typet::ASSUME || type==typet::GOTO)
+  if(is_assert() || is_assume() || is_goto())
     out << " (" << cond_value << ")";
+  else if(is_function_call() || is_function_return())
+    out << ' ' << identifier;
 
   if(hidden)
     out << " hidden";
 
-  out << "\n";
+  out << '\n';
+
+  if(is_assignment())
+  {
+    out << "  " << format(full_lhs)
+        << " = " << format(full_lhs_value)
+        << '\n';
+  }
 
   if(!pc->source_location.is_nil())
-    out << pc->source_location << "\n";
-
-  if(pc->is_goto())
-    out << "GOTO   ";
-  else if(pc->is_assume())
-    out << "ASSUME ";
-  else if(pc->is_assert())
-    out << "ASSERT ";
-  else if(pc->is_dead())
-    out << "DEAD   ";
-  else if(pc->is_other())
-    out << "OTHER  ";
-  else if(pc->is_assign())
-    out << "ASSIGN ";
-  else if(pc->is_decl())
-    out << "DECL   ";
-  else if(pc->is_function_call())
-    out << "CALL   ";
-  else
-    out << "(?)    ";
+    out << pc->source_location << '\n';
 
-  out << "\n";
+  out << pc->type << '\n';
```

Comment:
Why is this one changed but the switch statement at the top of this method isn't?